File: frontend/src/app/models/reservation.ts
export type ReservationStatus =
  'CONFIRMED' | 'CANCELLED';


export interface FootballField {

  id: number;

  name: string;

  active: boolean;

}


export interface UserResponse {

  id: number;

  nome: string;

  cognome: string;

  email: string;

  telefono: string;

  ruolo: string;

}


export interface Reservation {

  id: number;

  reservationDate: string;

  startTime: string;

  status: ReservationStatus;

  price: number;

  footballField: FootballField;

  user: UserResponse;

}

File: backend/src/main/java/com/fpc/football_booking/entity/Reservation.java
package com.fpc.football_booking.entity;

import com.fpc.football_booking.entity.enums.ReservationStatus;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;


@Entity
@Table(name = "reservations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @Column(nullable = false)
    private LocalDate reservationDate;



    @Column(nullable = false)
    private LocalTime startTime;



    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ReservationStatus status = ReservationStatus.CONFIRMED;

    @Column(nullable = false, precision = 8, scale = 2)
    private BigDecimal price;



    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "football_field_id",
            nullable = false
    )
    private FootballField footballField;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "user_id",
            nullable = false
    )
    private User user;

}
